keanu-python/keanu/tensor.py

```diff
@@ -72,8 +72,11 @@ def __get_tensor_from_scalar(scalar: primitive_types) -> JavaObject:
             raise NotImplementedError("Generic types in a ndarray are not supported. Was given {}".format(type(scalar)))
 
     @staticmethod
-    def _to_ndarray(java_tensor: JavaObject) -> numpy_types:
+    def _to_ndarray(java_tensor: JavaObject, primitive: bool = False) -> Any:
         if java_tensor.getRank() == 0:
-            return np.array(java_tensor.scalar())
+            if primitive == True:
+                return np.array(java_tensor.scalar()).item()
```

Comment:
Do you need to generate a numpy array at all here?

Reply:
Depends whether we want primitives or the numpy version of primitives.

I went for numpy but I can be easily swayed.

Reply:
Given we're explicitly returning a primitive here and have code to do it - I'd just return the non-numpy version?